### PR TITLE
vimPlugins.notational-fzf-vim: patch plugin to fix missing runtime dependency

### PR DESCRIPTION
**Pull request by brettlyons**

### CODE REVIEW

File: pkgs/misc/vim-plugins/overrides.nix
Comment (teto) [inline pkgs/misc/vim-plugins/overrides.nix]: I would say `with self` instead as we want to use the last defined packages.

Comment (timokau) [reply pkgs/misc/vim-plugins/overrides.nix]: If I recall correctly, `dependencies` is only supposed to be used for dependencies on other vim plugins. You probably want patch the plugin instead (hard-coding the ripgrep path or appending to `PATH`).

**Comment by timokau** [comment]
Did you run some kind of formatter? There are a lot of irrelevant changes in the diff.

**Comment by brettlyons** [comment]
Yeah, sorry, ran nix-fmt, reverting.

**Comment by brettlyons** [comment]
So, I'm not sure if this adds rg to the path, I do know that the plugin runs execute!("rg") rather than having it coded in, so adding ripgrep to PATH made the most sense to me.  Idk if buildinputs does that, or if something else is necessary?

**Comment by timokau** [comment]
No, `buildInputs` does not do that. It will only add `rg` to the `PATH` during *build time*, but not during *runtime*.

Unfortunately we don't currently have an easy way to do this at runtime. It could be done in combination with plugin management (simply adding an appropriate snippet to the generated vimrc), but we don't currently do that.

To make sure ripgrep can be found by the plugin, you will need to patch the plugin's source to contain an absolute path to ripgrep. You can do that `substituteInPlace` / `sed` / patch files. You can either directly modify the references to `rg` or add some code to prepend ripgrep to `PATH`.

**Comment by brettlyons** [comment]
python3 executable will need similar patch.  Don't merge yet :]

**Review by jonringer — COMMENTED** [review]
please clean up git history, should be a single commit

File: pkgs/misc/vim-plugins/overrides.nix
Comment (jonringer) [inline pkgs/misc/vim-plugins/overrides.nix]: does is specifically need pypy3? and not python3? I would find this unlikely as pypy is not a mainstream interpreter

Comment (brettlyons) [reply pkgs/misc/vim-plugins/overrides.nix]: Indeed, plugin docs says its optional.  Pulled pypy3 out as dependency and patched plugin to just use python3.

**Comment by brettlyons** [comment]
Okay, that's it, it builds and it works.  Potential improvement is to check for and use pypy3 if its available at runtime, using something like lib.strings.optionalString pkgs.pypy3 in the part of the definition in overrides where the python interpreter is set.

**Review by timokau — APPROVED** [review]
Looks good to me. Though keep in mind that this fix is pretty brittle, which is why appending to PATH is somewhat nicer. Still, this is a clear improvement and can be merged if @jonringer agrees.

File: pkgs/misc/vim-plugins/overrides.nix
Comment (jonringer) [inline pkgs/misc/vim-plugins/overrides.nix]: if these are being patched in, I don't you need them to be in buildInputs. Due to contextual strings, the related packages will be added to the derivation/runtime closure.
```suggestion
```

Comment (brettlyons) [reply pkgs/misc/vim-plugins/overrides.nix]: Okay.

**Review by jonringer — APPROVED** [review]

**Comment by timokau** [comment]
Thanks!

**Pull request merged**
